engine/engine_test.go

```diff
@@ -112,6 +113,71 @@ func TestQueryExplain(t *testing.T) {
 	}
 }
 
+func TestQueryAnalyze(t *testing.T) {
+	opts := promql.EngineOpts{Timeout: 1 * time.Hour}
+	series := storage.MockSeries(
+		[]int64{240, 270, 300, 600, 630, 660},
+		[]float64{1, 2, 3, 4, 5, 6},
+		[]string{labels.MetricName, "foo"},
+	)
+
+	start := time.Unix(0, 0)
+	end := time.Unix(1000, 0)
+
+	assertCPUTimeNonZero := func(t *testing.T, got *engine.AnalyzeOutputNode) {
+		if got != nil {
+			if got.OperatorTelemetry.CPUTimeTaken() <= 0 {
```

Comment:
Hmm, I think we should check for the children nodes too, and what structure those nodes have?